lib/markdown_videos/renderer.rb

```diff
@@ -13,42 +13,26 @@ def initialize(string, options = {})
     end
 
     def render
-      MarkdownVideos::SERVICES.keys.each do |service|
-        @string = render_service(service)
-      end
-      @string
-    end
+      @string.gsub(/!\[([^\]]*)\]\(([^)]+)\)/) do
```

Comment:
But gsub doesn't change `@string`, does it? Usually you don't want a variable to represent different things at different points, I would say.